File: project2/Antelope.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;

class Antelope extends Animal {
    private static final Random random = new Random();
    
    public Antelope(int x, int y, World w) {
        super(4, 4, x, y, w);
    }
    
    @Override
    public char draw() {
        return 'A';
    }
    
    @Override
    public Animal clone(int x, int y) {
        return new Antelope(x, y, world);
    }
    
    @Override
    public void action() {
        int[][] directions = {{0, -2}, {0, 2}, {-2, 0}, {2, 0}};
        ArrayList<int[]> dirs = new ArrayList<>();
        for (int[] dir : directions) dirs.add(dir);
        Collections.shuffle(dirs);
        
        for (int[] dir : dirs) {
            int nx = x + dir[0];
            int ny = y + dir[1];
            
            if (nx >= 0 && ny >= 0 && nx < 40 && ny < 20) {
                Organism target = world.grid[ny][nx];
                if (target != null) collision(target);
                if (alive) {
                    world.grid[y][x] = null;
                    x = nx;
                    y = ny;
                    world.grid[y][x] = this;
                }
                break;
            }
        }
        age++;
    }
}

File: project3/antelope.py
from animal import Animal
import random

class Antelope(Animal):
    def __init__(self, x, y, world):
        super().__init__(4, 4, x, y, world)
    
    def action(self):
        # Antelope moves by 2 cells at once
        direction = random.randint(0, 3)
        nx, ny = self.x, self.y
        
        if direction == 0 and self.y > 1:  # Up
            ny -= 2
        elif direction == 1 and self.y < self.HEIGHT - 2:  # Down
            ny += 2
        elif direction == 2 and self.x > 1:  # Left
            nx -= 2
        elif direction == 3 and self.x < self.WIDTH - 2:  # Right
            nx += 2
        
        if self.world.grid[ny][nx] is not None:
            self.collision(self.world.grid[ny][nx])
            
        if self.alive and self.move:
            self.world.grid[self.y][self.x] = None
            self.x, self.y = nx, ny
            self.world.grid[self.y][self.x] = self
        
        self.age += 1
        self.move = True
    
    def draw(self):
        return 'A'
    
    def clone(self, x, y):
        return Antelope(x, y, self.world)
